src/Data/Profunctor/Linear.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Data.Profunctor.Linear
+  ( Profunctor(..)
+  , SymmetricMonoidal(..)
+  , Monoidal(..)
+  , Strong(..)
+  ) where
+
+import Data.Void
+import Prelude.Linear hiding (swap)
+
+-- TODO: write laws
+
+class Profunctor (arr :: * -> * -> *) where
+  {-# MINIMAL dimap | lmap, rmap #-}
+
+  dimap :: (s ->. a) -> (b ->. t) -> a `arr` b -> s `arr` t
+  dimap f g x = lmap f (rmap g x)
+  {-# INLINE dimap #-}
+
+  lmap :: (s ->. a) -> a `arr` t -> s `arr` t
+  lmap f = dimap f id
+  {-# INLINE lmap #-}
+
+  rmap :: (b ->. t) -> s `arr` b -> s `arr` t
+  rmap = dimap id
+  {-# INLINE rmap #-}
+
+-- TODO: Move to a dedicated module
```

Comment:
Let's make a `Data.Bifunctor.Linear` module. Add a (linear) `Bifunctor` type class there. And make `SymmetricMonoidal` a super class.